### ISSUE TITLE
getAll retruns 0 if there is only one provider 

### ISSUE THREAD

**Issue by subspider** [InsertKoinIO/koin#832]
**Describe the bug**
data class CommunityServiceLogic(
    private val services: List<CommunityApiService>
)

if we only provide one CommunityApiService let's say " 

`single<CommunityApiService> { CommunityApiServiceImple(get()) }`
 
`factory { CommunityServiceLogic(getAll()) }` **this will not work** 

`factory { CommunityServiceLogic(get()) } `  **this will work** 

**Expected behavior**

getAll() should reaturn at least one element if it is provided   

**Koin project used and used version (please complete the following information):**

```
            'koin'                     : [
                    core     : "org.koin:koin-core:${versions.koin}",
                    android  : "org.koin:koin-android:${versions.koin}",
                    scope    : "org.koin:koin-androidx-scope:${versions.koin}",
                    viewmodel: "org.koin:koin-androidx-viewmodel:${versions.koin}",
                    test     : "org.koin:koin-test:${versions.koin}"
            ],
```
**KOIN VERSION =  2.1.6**

**Comment by arnaudgiuliani** [comment]
Fixed in 2.2.0. not reproduced in 2.2.0-rc-4

**Comment by subspider** [comment]
@arnaudgiuliani thanks
